lib/webdriver.js

```diff
@@ -105,12 +105,17 @@ Webdriver.prototype._init = function() {
     if(err) { return cb(err); }
 
     var resData;
-    // retrieving session
+    var jsonData;
+
     try{
-      var jsonData = JSON.parse(data);
+      jsonData = JSON.parse(data);
       if( jsonData.status === 0 ){
+        // retrieving session via old JSON wire protocol
         _this.sessionID = jsonData.sessionId;
         resData = jsonData.value;
+      } else {
+        // retrieving session via the new W3C protocol
+        _this.sessionID = jsonData.value && jsonData.value.sessionId;
```

Comment:
this will set sessionId to false if `jsonData.value` doesn't exist; should we throw instead?

Reply:
If `jsonData.value` is `undefined`, it will set sessionID to `undefined`.  Then the existing code below will trigger "The environment you requested was unavailable."